Common/src/Protocol/SignallingProtocol.ts

```diff
@@ -24,27 +24,48 @@ import * as MessageHelpers from '../Messages/message_helpers';
  *   out:
  *      Emitted when sending a message out on the transport. Similar to 'message' but
  *      only for when messages are sent from this endpoint. Useful for debugging.
+ *
+ *   timeout:
+ *      Emitted when a ping fails to receive a pong in time and the connection is
+ *      assumed to be dead. This is emitted after the connection is closed.
  */
 export class SignallingProtocol extends EventEmitter {
     static get SIGNALLING_VERSION(): string {
         return '1.2.1';
     }
 
+    static readonly PING_TIMEOUT: number = 30 * 1000;
```

Comment:
This feels like it should be configurable in:

1) frontend PS code
2) in SS config

The default seems good, but I can imagine people may have cases where they need it set to other values for some particular reason.